src/commands/cleanup.ts

```diff
@@ -208,17 +211,20 @@ export default class extends Command {
         const components: Array<Component> = []
         const paths: Array<string> = []
         for (let item of output) {
-            let trimmedPath : string = item.assets[0].path.split("/").slice(0, pathDepth).join("/")
-            components.push(new Component()
-                .withDate(item.assets[0].lastModified)
-                .withId(item.id)
-                .withVersion(item.version)
-                .withPath(trimmedPath)
-            )
-            if (!paths.includes(trimmedPath)) {
-                paths.push(trimmedPath)
+            // skip if first asset has no path
+            if(item.assets[0].hasOwnProperty('path')) {
```

Comment:
I'd be more stricter:
```suggestion
            if(item.assets.length > 0 && item.assets[0].path) {
```
(also you can directly ask for the property. It will return "undefined" which is falsey. No need to use hasOwnProperty)

Reply:
wouldn't that not already trigger that
TypeError: Cannot read properties of undefined (reading 'path')
error?

Reply:
No, the error was triggered in the next line where you used .split on the non-existent path property.